src/language-markdown/embed.js

```diff
@@ -52,6 +52,16 @@ function embed(path, print, textToDoc, options) {
         ),
         hardline,
       ];
+    case "esComment":
+      return (
+        "{" +
+        textToDoc(
+          "/* " + node.value + " */",
+          { parser: "babel" },
+          { stripTrailingHardline: true }
+        ) +
+        "}"
+      );
```

Comment:
I’m not sure why the MDX stuff is in `embed`, but that’s why I added it here

Reply:
I think this is not the right place to add it, it should be here https://github.com/prettier/prettier/blob/5aabd709505e6e858396586d2811b98fb06e40fc/src/language-markdown/printer-markdown.js#L75

When we disable embed language, it will throw.

[Playground link](https://deploy-preview-11563--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBBABMA9AKnQKQGUBaAZxgE8AbOdSAW3oRnRywF86JHmA6AHSiDseVFAgwAFnABOBEuWq0GTWKw4gANCAgAHGAEtopZKACGMmRADuABQsITKM1WtmKJ7QCMZZsAGs4GEIzJgAZAyg4ZAAzF1I4b18AoMJdP0iAc2QYGQBXRJA4ei84ABMy8rCzKEy8s0y4ADEIGXozGENa5B0YmK0QSRh6KgB1SQN4UnSwOEJHSYMAN0mKHrBSTxBIhJkYW19M9tj4woArUgAPQiyaAEU8iWikOKoE7XSZXZ76MsuB3QySIwUYGMpSZAADgADB8rAlRr5dD1AXBdktotoAI6PeAHPROEBmUjEKLlcoDGRwHEGKkHBrHF6nbQJegGHL5QqkW5wB5PE5vQowMxeUHgyTIABM2lyZgMVCyAGFuIyiqQAKwDPIJAAqIqcr3eICWBQAklBKrBCGAgfoxGVCJQaAKEux2EA)

Reply:
Pushed a fix to also add it to the normal printer. It looks like `embed` is for formatting embedded languages (so JS in MD), but not always available, hence it’s in two places?

Reply:
Given they print the same result, I think we can remove this one.

Please add a test in https://github.com/prettier/prettier/tree/main/tests/format/mdx/embedded-language-formatting

Reply:
Does the embedded parser perhaps break lines in comments or in other ways format them? 🤔

Reply:
Your current implementation just call the babel parser to print a JS block comment, and we print single comment as-it-is in JS printer.

The real question is does this comment support multiple line? And does it support comments that all lines starts with `*`? [We called it "IndentableBlockComment"](https://github.com/prettier/prettier/blob/5aabd709505e6e858396586d2811b98fb06e40fc/src/language-js/print/comment.js#L53), if it does, there will be differences (you didn't add test for it).

Reply:
Hmm okay. I removed it.

I think the primary goal right now is to add support for it. Then in the future, we can improve the parser, including the MDX part, by updating everything to latest remark?

Reply:
Okay, I was going to ask you about that... I understand that's too much for you, but I failed to update `remark-parse`(#9397). It will be great, if you can help us for the upgrade sometime later.

Reply:
My goal is to get MDX to an RC in 2 weeks (essentially porting xdm back into it and adding docs). So we do have some time (before but also probably after it).

I can help, yeah. All of unified is now ESM-only. I believe prettier is bundling dependencies so that shouldn’t be a problem for end users. Can ESM be setup internally in prettier/src/language-markdown?

Reply:
Unfortunately, no. We still don't the problem, but we can't make ES Module work for the tests, even a simple esm file...  #11235

I guess we'll have to wait until Jest have better support for modules.

Reply:
Also, our migration to Pure ESM can be tracked at #10157

Reply:
can the markdown language be compiled first before tests?